### DISCUSSION TITLE
Embed music or video

### DISCUSSION THREAD

**Discussion by camilajenny** [zim-desktop-wiki/zim-desktop-wiki#1839] [Ideas]
It would be nice if we were able to embed a video or a piece of music inside a page so that we can just press play and play it without the need to use a third-party application.

Let me visualize my idea:
![image](https://user-images.githubusercontent.com/17968722/43967039-bea66054-9cc3-11e8-9197-81365c456d9f.png)

**Comment by quesada** [comment]
Agreed. This feature request: "insert video on pages" is more important as time passes. Zim is often used as documentation, and the web has moved towards video more and more. It's becoming very easy to record short videos and insert them on documentation systems.

**Comment by introt** [comment]
Re: @daxime over at #1828 

> If you dont want to add a video player, maybe you can make it launch the default player of the system.

This can be achieved right now manually(/with a custom tool): embed the video thumbnail as an image (with a "play" symbol superimposed) and link it to the video file; clicking on the image will open the video in the default player.

A simple "actual implementation" could be similar; when dragging video files into Zim, instead of creating a text link it could create an object similar to ImageGenerator plugins- something that looks like an image in the editor, but could automatically be turned into eg. a HTML5 media element by the exporter.

**Comment by quesada** [comment]
Pasting a youtube link could generate a player embedded in the text (or the  video thumbnail as an image (with a "play" symbol superimposed) and link it to the video file). We are in an era where web links are treated as 1st class citizens (Slack created the 'preview', and every tool after that copied them)

**Reply by introt** [reply]
From the [markdown blueprint](https://github.com/zim-desktop-wiki/zim-desktop-wiki/wiki/BlueprintMarkdown):

> Use inline object to show the html - do not try to parse it

If we're going to have some sort of inline html objects (following the current trend set by the generator plugins, they'd likely use a headless browser to create PNGs lol), using those for "web stuff" should be easy.
